### PR TITLE
Weekend challenge submission

### PR DESCRIPTION
**Pull request by adamjohnsnow**

### CODE REVIEW

File: spec/weather_spec.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require './lib/weather'
+
+describe Weather do
+  let(:airport) { double :airport, name: 'the Airport' }
+  it '#weather can be #sunny or #stormy' do
+    weather = Weather.new(airport)
+    expect(weather.condition).to eq("Sunny").or(eq("Stormy"))
```
Comment (neoeno) [inline spec/weather_spec.rb]: This would pass if it always returned `"Sunny"`, right?

File: spec/plane_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+require './lib/plane'
+
+describe Plane do
+
+  context 'when airport is sunny' do
+    let(:airport) { double :airport, name: "Test", full?: false, runway: [], land_plane: '', takeoff_plane: '' }
```
Comment (neoeno) [inline spec/plane_spec.rb]: Two of these attributes you've given to `double` aren't required — can you tell me which and why?

File: spec/plane_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    let(:weather) { double :weather, condition: "Sunny", check_weather: "Sunny"}
```
Comment (neoeno) [inline spec/plane_spec.rb]: And one of these attributes — but perhaps that's an easier question!

File: spec/airport_spec.rb [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    end
+
+    it "plane cannot #land_at a #full #airport" do
+      subject.runway = ["a", "a", "a", "a", "a", "a", "a", "a", "a", "a"]
+      expect { subject.landing_check(plane) }.to raise_error "Cannot land, airport full"
+    end
+
+  end
+
+  context 'when #airport is #stormy' do
```
Comment (neoeno) [inline spec/airport_spec.rb]: These `#` have a specific meaning in RSpec — to refer to instance methods — it's clever the way you've extended them here but it probably decreases readability sadly!

File: spec/airport_spec.rb [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    it "has a #runway" do
+      expect(subject.runway.class).to eq(Array)
+    end
+
+    it "can be #full" do
+      expect(subject).to respond_to(:full?)
+    end
+
+    it "recognised #runway with 10 as #full" do
+      subject.runway = ["a", "a", "a", "a", "a", "a", "a", "a", "a", "a"]
```
Comment (neoeno) [inline spec/airport_spec.rb]: Maybe give `["a"] * 10` a go :)

File: lib/airport.rb [hunk 6]
```diff
@@ -0,0 +1,47 @@
+require_relative 'plane'
+require_relative 'weather'
+
+class Airport
+
+  attr_accessor  :current_weather, :name, :runway, :capacity, :landed_at
```
Comment (neoeno) [inline lib/airport.rb]: Would it be possible to make some or all of these private? Any reason why we'd do that?

File: lib/airport.rb [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+  def takeoff_check(plane)
+    @current_weather = generate_weather
+    fail "Cannot takeoff, weather at #{name} stormy" if @current_weather.condition == "Stormy"
+  end
+
+  def landing_check(plane)
+    @current_weather = generate_weather
+    fail "Cannot land, weather at #{name} stormy" if @current_weather.condition == "Stormy"
+    fail "Already landed at #{name}" unless plane.landed_at.nil?
```
Comment (neoeno) [inline lib/airport.rb]: Here, `Airport` looks at one of `Plane`'s instance variables to see if it's nil. This feels like maybe something `Plane` should be doing.

File: spec/weather_spec.rb (hunk 1 above)
Comment (adamjohnsnow) [inline spec/weather_spec.rb]: Yes, this was my attempt at checking that one of the two conditions were always returned, yet could still be random.

File: spec/plane_spec.rb (hunk 2 above)
Comment (adamjohnsnow) [inline spec/plane_spec.rb]: Ah yes. I think I was just trying to replicate Airport as much as possible in desperation.

File: spec/airport_spec.rb (hunk 5 above)
Comment (adamjohnsnow) [inline spec/airport_spec.rb]: Well I'll be damned!!

File: spec/airport_spec.rb (hunk 4 above)
Comment (adamjohnsnow) [inline spec/airport_spec.rb]: Ah - I thought they referred to variables too.

File: lib/airport.rb (hunk 7 above)
Comment (adamjohnsnow) [inline lib/airport.rb]: Gotcha - my understanding of SRP and delegation is much better this weekend... I hope.
Thanks for your time on this, I hope you are able to have a look at this weekend's challenge and that I have shown some improvement.
Thanks

**Pull request closed without merging**